admin_manual/enterprise/user_management/saml_2.0_sso.rst

```diff
@@ -0,0 +1,200 @@
+To do so, use ``adfs2fed.php``, as in the following command:
+
+.. code-block:: console
+
+  php apps/user_shibboleth/tools/adfs2fed.php \
+     https://hostname.<AD-Domain>/FederationMetadata/2007-06/FederationMetadata.xml \
+     <AD-Domain> > /etc/shibboleth/filtered-metadata.xml
+
+Configure mod_shib
+~~~~~~~~~~~~~~~~~~
+
+Next, you need to configure mod_shib. 
+To do this, in ``/etc/shibboleth/shibboleth2.xml``:
+
+1. Use the URL of the ownCloud instance as the ``entityID`` in the ``ApplicationDefaults``, e.g.,
+
+   .. code:: xml
+
+    <ApplicationDefaults entityID="https://cloud.hostname.com/login/saml" REMOTE_USER="eppn upn">
+
+   .. note::
+
+    ``https://cloud.hostname.com/login/saml`` is just an example adjust to your needs.
+
+2. Configure the SSO to use the ``entityID`` from the ``filtered-metadata.xml``, e.g.,
+
+   .. code:: xml
+
+    <SSO entityID="https://sts.hostname.tld/11537aff-6d87-4bb1-ae81-d26948b8ea28/">
+        SAML2
+    </SSO>
+
+3. Configure an XML ``MetadataProvider`` with the local ``filtered-metadata.xml`` file:
+
+   .. code:: xml
+
+    <MetadataProvider type="XML" file="/etc/shibboleth/filtered-metadata.xml"/>
+
+Further Reading
+^^^^^^^^^^^^^^^
+    
+- https://wiki.shibboleth.net/confluence/display/SHIB2/NativeSPApplication#NativeSPApplication-BasicConfiguration(Version2.4andAbove)
+- https://wiki.shibboleth.net/confluence/display/SHIB2/NativeSPServiceSSO
+- https://wiki.shibboleth.net/confluence/display/SHIB2/NativeSPMetadataProvider#NativeSPMetadataProvider-XMLMetadataProvider
+
+Metadata Available
+~~~~~~~~~~~~~~~~~~
+
+Under ``https://cloud.hostname.tld/Shibboleth.sso/Metadata`` mod_shib exposes the Metadata that is needed by ADFS to add the SP as a Relying party.
+
+ADFS
+-----
+
+This part needs to be done by an ADFS administrator. 
+Let him do his job while you continue with the Apache configuration below. 
+
+Add a Relying Party Using Metadata
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+See `AD FS 2.0 Step-by-Step Guide <https://technet.microsoft.com/en-us/library/gg317734(v=ws.10).aspx>`_ step 2.
+
+Configure ADFS to send the userPrincipalName in the SAML token
+---------------------------------------------------------------
+
+If you have control over ADFS make it send the ``UPN`` as ``eppn`` by adding the following custom claim rule:
+
+.. code-block:: xml
+
+   c:[Type == "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/upn"]
+   => issue(Type = "urn:oid:1.3.6.1.4.1.5923.1.1.1.6", Value = c.Value, Properties["http://schemas.xmlsoap.org/ws/2005/05/identity/claimproperties/attributename"] = "urn:oasis:names:tc:SAML:2.0:attrname-format:uri");
+
+This will allow the default configuration of mod_shib to pick up the ``userPrincipalName`` and present it as the ``eppn`` to the SP.
+If something prevents you from sending a custom rule send the ``userPrincipalName`` and change mod_shibs ``attribute-map.xml`` to
```

Comment:
- [ ] describe this as two distinct options:
1. AD sends UPN as EPPN:
    - mod_shib maps eppn to environment as eppn (uses scope filter, scope added in `filtered-metadata.xml` above),
    - user_shibboleth reads default `eppn` environment variable
2. AD sends UPN as UPN
    - mod_ship maps upn to environment as upn (no scope filter),
    - user_shibboleth needs to be configured to read `upn` environment variable